src/unittest/test_socket.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+Minetest
+Copyright (C) 2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "test.h"
+
+#include "log.h"
+#include "socket.h"
+#include "settings.h"
+
+class TestSocket : public TestBase {
+public:
+	TestSocket()
+	{
+		if (INTERNET_SIMULATOR == false)
+			TestManager::registerTestModule(this);
+	}
+
+	const char *getName() { return "TestSocket"; }
+
+	void runTests(IGameDef *gamedef);
+
+	void testIPv4Socket();
+	void testIPv6Socket();
+
+	static const int port = 30003;
+};
+
+static TestSocket g_test_instance;
+
+void TestSocket::runTests(IGameDef *gamedef)
+{
+	TEST(testIPv4Socket);
+
+	if (g_settings->getBool("enable_ipv6"))
```

Comment:
The order in which these test units are run depend totally on the linker used to build minetest.  On my machine it's reverse alphabetical order.

My rationale for this being okay is that TestSettings will *eventually* be run, and upon noticing this, you can figure that something is probably going wrong with the settings if this is failing too.

Besides, I plan on removing any g_settings usage in unit tests.  Here, this specific check is redundant since the unit test effectively skips if IPv6 support isn't detected on the host, which I believe is the correct behavior.